samples/show-device-location/src/main/java/com/esri/arcgismaps/sample/showdevicelocation/screens/ShowDeviceLocationScreen.kt

```diff
@@ -0,0 +1,234 @@
+import com.arcgismaps.ArcGISEnvironment
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.arcgismaps.toolkit.geoviewcompose.rememberLocationDisplay
+import com.esri.arcgismaps.sample.showdevicelocation.components.ShowDeviceLocationViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun ShowDeviceLocationScreen(sampleName: String) {
+    val mapViewModel: ShowDeviceLocationViewModel = viewModel()
+
+    val context = LocalContext.current
+
+    // some parts of the API require an Android Context to properly interact with Android system
+    // features, such as LocationProvider and application resources
+    ArcGISEnvironment.applicationContext = context.applicationContext
+
+    // Create and remember a location display with a recenter auto pan mode.
+    val locationDisplay = rememberLocationDisplay()
+
+    // this variable controls the visibility of the dropdown menu
+    var showDropDownMenu by remember { mutableStateOf(false) }
+
+    RequestPermissions(
+        context = context,
+        onPermissionsGranted = {
+            mapViewModel.toggleLocationTracking(true, locationDisplay)
+        }
+    )
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+            ) {
+                MapView(
+                    modifier = Modifier
+                        .fillMaxSize(),
+                    arcGISMap = mapViewModel.arcGISMap,
+                    locationDisplay = locationDisplay,
+                )
+
+                Card (
+                    modifier = Modifier
+                        .align(Alignment.BottomStart)
+                        .padding(horizontal = 20.dp, vertical = 70.dp)
+                        .border(2.dp, MaterialTheme.colorScheme.primary, RoundedCornerShape(20.dp))
+                        .width(150.dp)
+                        .height(50.dp),
+                    shape = RoundedCornerShape(20.dp),
+                    elevation = CardDefaults.cardElevation(
+                        defaultElevation = 4.dp
+                    ),
+                ) {
+                    Row(
+                        modifier = Modifier.fillMaxSize(),
+                        verticalAlignment = Alignment.CenterVertically,
+                        horizontalArrangement = Arrangement.SpaceEvenly
+                    ) {
+                        Text(text = "Off", fontSize = 22.sp)
+                        Switch(
+                            checked = mapViewModel.isLocationTrackingEnabled,
+                            onCheckedChange = {mapViewModel.toggleLocationTracking(it, locationDisplay)}
```

Comment:
Formatting and removal of `it` (which causes a shadowing warning):
```suggestion
                            onCheckedChange = { isChecked ->
                                mapViewModel.toggleLocationTracking(
                                    isChecked,
                                    locationDisplay
                                )
                            }